pkg/kubectl/cmd/set/set_probe.go

```diff
@@ -0,0 +1,457 @@
+		errors = append(errors, fmt.Errorf("one or more resources must be specified as <resource> <name> or <resource>/<name>"))
+	}
+	if !o.Readiness && !o.Liveness {
+		errors = append(errors, fmt.Errorf("you must specify one of --readiness or --liveness or both"))
+	}
+	count := 0
+	if o.Command != nil {
+		count++
+	}
+	if len(o.OpenTCPSocket) > 0 {
+		count++
+	}
+	if len(o.HTTPGet) > 0 {
+		count++
+	}
+
+	switch {
+	case o.Remove && count != 0:
+		errors = append(errors, fmt.Errorf("--remove may not be used with any flag except --readiness or --liveness"))
+	case count > 1:
+		errors = append(errors, fmt.Errorf("you may only set one of --get-url, --open-tcp, or command"))
+	case len(o.OpenTCPSocket) > 0 && intOrString(o.OpenTCPSocket).IntVal > 65535:
+		errors = append(errors, fmt.Errorf("--open-tcp must be a port number between 1 and 65535 or an IANA port name"))
+	}
+	if o.FailureThreshold != nil && *o.FailureThreshold < 1 {
+		errors = append(errors, fmt.Errorf("--failure-threshold may not be less than one"))
+	}
+	if o.SuccessThreshold != nil && *o.SuccessThreshold < 1 {
+		errors = append(errors, fmt.Errorf("--success-threshold may not be less than one"))
+	}
+	if o.InitialDelaySeconds != nil && *o.InitialDelaySeconds < 0 {
+		errors = append(errors, fmt.Errorf("--initial-delay-seconds may not be negative"))
+	}
+	if o.TimeoutSeconds != nil && *o.TimeoutSeconds < 0 {
+		errors = append(errors, fmt.Errorf("--timeout-seconds may not be negative"))
+	}
+	if o.PeriodSeconds != nil && *o.PeriodSeconds < 0 {
+		errors = append(errors, fmt.Errorf("--period-seconds may not be negative"))
+	}
+	return utilerrors.NewAggregate(errors)
+}
+
+func (o *ProbeOptions) Run() error {
+	allErrs := []error{}
+	patches := CalculatePatches(o.Infos, o.Encoder, func(info *resource.Info) ([]byte, error) {
+		transformed := false
+		_, err := o.UpdatePodSpecForObject(info.Object, func(spec *api.PodSpec) error {
+			containers, _ := selectContainers(spec.Containers, o.ContainerSelector)
+			if len(containers) == 0 {
+				if _, err := fmt.Fprintf(o.Err, "warning: %s/%s does not have any containers matching %q\n", info.Mapping.Resource, info.Name, o.ContainerSelector); err != nil {
+					return err
+				}
+				return nil
+			}
+			// perform updates
+			transformed = true
+			for _, container := range containers {
+				o.updateContainer(container)
+			}
+			return nil
+		})
+		if transformed && err == nil {
+			return runtime.Encode(o.Encoder, info.Object)
+		}
+		return nil, err
+	})
+
+	for _, patch := range patches {
+		info := patch.Info
+		if patch.Err != nil {
+			fmt.Fprintf(o.Err, "error: %s/%s %v\n", info.Mapping.Resource, info.Name, patch.Err)
+			continue
+		}
+
+		if string(patch.Patch) == "{}" || len(patch.Patch) == 0 {
+			if _, err := fmt.Fprintf(o.Out, "%s %q was not changed\n", info.Mapping.Resource, info.Name); err != nil {
+				return err
+			}
+			continue
+		}
+		if o.PrintObject != nil && (o.Local || o.DryRun) {
```

Comment:
I think an error should be returned if `local` or `dryRun` are specified but `o.PrintObject` is still `nil`

Reply:
i feel it is redundance. now have check nil, so will not cause panic, i 'd more like keep origin. because other `set subcommand` alse not have logged this~~